kernel/tests/read.rs

```diff
@@ -488,8 +496,63 @@ fn column_ordering_and_projection() -> Result<(), Box<dyn std::error::Error>> {
     read_table_data(
         "./tests/data/basic_partitioned",
         Some(&["a_float", "number"]),
+        None,
         expected,
     )?;
 
     Ok(())
 }
+
+#[test]
+fn predicate_on_number() -> Result<(), Box<dyn std::error::Error>> {
+    let expected = vec![
+        "+---------+--------+",
+        "| a_float | number |",
+        "+---------+--------+",
+        "| 1.1     | 1      |",
+        "| 2.2     | 2      |",
+        "| 3.3     | 3      |",
+        "+---------+--------+",
+    ];
+    read_table_data(
+        "./tests/data/basic_partitioned",
+        Some(&["a_float", "number"]),
+        Some(Expression::Column("number".to_string()).lt(Expression::Literal(Scalar::Long(4)))),
+        expected,
+    )?;
+    Ok(())
+}
+
+#[test]
+fn predicate_on_number_with_not_null() -> Result<(), Box<dyn std::error::Error>> {
+    let expected = vec![
+        "+---------+--------+",
+        "| a_float | number |",
+        "+---------+--------+",
+        "| 1.1     | 1      |",
+        "| 2.2     | 2      |",
+        "+---------+--------+",
+    ];
+    read_table_data(
+        "./tests/data/basic_partitioned",
+        Some(&["a_float", "number"]),
+        Some(Expression::and(
+            Expression::Column("number".to_string()).is_null().not(),
```

Comment:
Should we [`impl std::ops::Not for Expression`](https://doc.rust-lang.org/std/ops/trait.Not.html)? it would allow
```suggestion
            !Expression::column("number").is_null(),
```
(but also might be more complex/confusing than it's worth)